sdk/cosmosdb/cosmos/src/globalEndpointManager.ts

```diff
@@ -126,13 +126,14 @@ export class GlobalEndpointManager {
       ? this.readableLocations
       : this.writeableLocations;
 
+    this.refreshUnavailableLocations(locations);
```

Comment:
This is looping on all locations every time a request needs to be routed. In other SDKs we have a periodic (5min) background refresh of the account information, where this process happens (checking if unavailability should be lifted).

Do we have the background refresh in this SDK? (references https://github.com/Azure/azure-cosmos-dotnet-v3/blob/master/Microsoft.Azure.Cosmos/src/Routing/GlobalEndpointManager.cs#L27, https://github.com/Azure/azure-cosmos-dotnet-v3/blob/master/Microsoft.Azure.Cosmos/src/Routing/LocationCache.cs#L490)

Reply:
I second not looping through all endpoints at every request. @topshot99 we already have a periodic thread `backgroundRefreshEndpointList` which refreshes endpoints, can we use that OR create a similar thread.

Reply:
@topshot99 The call starts on the Global Endpoint https://github.com/Azure/azure-cosmos-dotnet-v3/blob/master/Microsoft.Azure.Cosmos/src/Routing/GlobalEndpointManager.cs#L435, once every 5 minutes, the LocationCache's OnDatabaseAccountRead is called (https://github.com/Azure/azure-cosmos-dotnet-v3/blob/77e3aa44c5704f82fcfd6e1028327e4259b40f5f/Microsoft.Azure.Cosmos/src/Routing/GlobalEndpointManager.cs#L543), which calls the UpdateLocationCache (https://github.com/Azure/azure-cosmos-dotnet-v3/blob/77e3aa44c5704f82fcfd6e1028327e4259b40f5f/Microsoft.Azure.Cosmos/src/Routing/LocationCache.cs#L192-L199).

I am not super familiar with the Node SDK, but some of this logic might have been simplified into a single file.